lang_qc/endpoints/pacbio_wells.py

```diff
@@ -159,90 +154,13 @@ def grab_wells_with_status(
             raise Exception("An unknown filter was passed.")
 
 
-def group_wells_into_inbox_results(well_db_records: List[PacBioRunWellMetrics]):
-
-    # Group the wells by run.
-    grouped_wells = {}
-
-    for well_metrics in well_db_records:
-        run_name = well_metrics.pac_bio_run_name
-        if run_name not in grouped_wells:
-            grouped_wells[run_name] = []
-        grouped_wells[run_name].append(well_metrics)
-
-    # Next, convert the grouped wells into result entries.
-    def make_inbox_result_entry(
-        well_metrics: Tuple[str, List[PacBioRunWellMetrics]]
-    ) -> InboxResultEntry:
-        """Construct an InboxResultEntry from a list of wells"""
-        run_name, well_metrics = well_metrics  # unpack the grouped entries
-        time_start = well_metrics[0].run_start
-        time_complete = well_metrics[0].run_complete
-
-        wells = sorted(
-            [
-                WellInfo(
-                    label=well.well_label,
-                    start=well.well_start,
-                    complete=well.well_complete,
-                )
-                for well in well_metrics
-            ],
-            key=attrgetter("label"),
-        )
-
-        return InboxResultEntry(
-            run_name=run_name,
-            time_start=time_start,
-            time_complete=time_complete,
-            wells=wells,
-        )
-
-    unsorted_results = map(make_inbox_result_entry, grouped_wells.items())
-
-    # Finally, sort the wells by most recently completed well.
-    def get_max_date(entry):
-        """Get the most recent well completion date for wells in an entry."""
-
-        return max(entry.wells, key=attrgetter("complete")).complete
-
-    output = sorted(
-        unsorted_results,
-        key=get_max_date,
-        reverse=True,
-    )
-
-    return output
-
-
-@router.get(
-    "/inbox",
-    response_model=InboxResults,
-    responses={400: {"description": "Bad Request. Invalid weeks parameter."}},
-)
-def get_inbox(
-    weeks: int = 2, db_session: Session = Depends(get_mlwh_db)
-) -> InboxResults:
-    """Get inbox of PacBio runs.
-
-    Fetches from the last 2 weeks by default."""
-
-    if weeks < 0:
-        raise HTTPException(
-            status_code=400, detail="Bad Request. Invalid weeks parameter."
-        )
-
-    results: List[PacBioRunWellMetrics] = grab_recent_wells_from_db(weeks, db_session)
-
-    return group_wells_into_inbox_results(results)
-
-
 @router.get("/wells", response_model=FilteredInboxResults, tags=["Well inbox"])
 def get_wells_filtered_by_status(
     qc_status: QcStatusEnum = QcStatusEnum.INBOX,
+    weeks: int = 1,
     qcdb_session: Session = Depends(get_qc_db),
     mlwh_session: Session = Depends(get_mlwh_db),
 ):
 
-    wells, states = grab_wells_with_status(qc_status, qcdb_session, mlwh_session)
+    wells, states = grab_wells_with_status(qc_status, qcdb_session, mlwh_session, weeks)
```

Comment:
Can you have a think over whether `grab_wells_with_status` or `pack_wells_and_states` need any exception trapping here? At the moment they will create 500s for everything that can go wrong.

Reply:
We can validate number of weeks - should be positive. Other than that -- cannot think of anything. I think it's worth testing that the code does not fail if there are no filtered data. Should return an empty list. Probably can add this test before changing the nature of the list returned.